File: examples/python-repo/ecommerce/domain/inventory/inventory.py
class Inventory:
    def __init__(self):
        self._stocks = {}

    def add_stock(self, stock):
        self._stocks[stock.product_id] = stock

    def get_stock(self, product_id):
        stock = self._stocks.get(product_id)
        if stock is None:
            raise ValueError(f"no stock record for product {product_id}")
        return stock

    def is_available(self, product_id, quantity):
        stock = self._stocks.get(product_id)
        return stock is not None and stock.has_enough(quantity)

    def reserve(self, product_id, quantity):
        self.get_stock(product_id).reserve(quantity)

    def restock(self, product_id, quantity):
        self.get_stock(product_id).restock(quantity)

File: examples/java-repo/src/main/java/com/example/ecommerce/domain/inventory/Inventory.java
package com.example.ecommerce.domain.inventory;

import java.util.HashMap;
import java.util.Map;

public class Inventory {

    private final Map<Long, Stock> stocks;

    public Inventory() {
        this.stocks = new HashMap<>();
    }

    public void addStock(Stock stock) {
        stocks.put(stock.getProductId(), stock);
    }

    public Stock getStock(long productId) {
        Stock stock = stocks.get(productId);
        if (stock == null) {
            throw new IllegalArgumentException("no stock record for product " + productId);
        }
        return stock;
    }

    public boolean isAvailable(long productId, int quantity) {
        Stock stock = stocks.get(productId);
        return stock != null && stock.hasEnough(quantity);
    }

    public void reserve(long productId, int quantity) {
        getStock(productId).reserve(quantity);
    }

    public void restock(long productId, int quantity) {
        getStock(productId).restock(quantity);
    }
}
